### PR TITLE
Workspace cleanup

### PR DESCRIPTION
**Pull request by justin8**
@fjakobs

Ready to go. For testing, make sure to build using https://github.com/c9/newclient/pull/12435

### CODE REVIEW

**Comment by modius** [comment]
A worthy mission!

File: workspace/Dockerfile [hunk 1]
```diff
@@ -29,65 +34,46 @@ RUN apt-get install -y bash console-setup sudo wget curl git python man-db && \
...
-        nvm install iojs-v3.3.1 && \
-        nvm install 4.2.4 && \
-        nvm install 5.4.1 && \
-        nvm alias default 4.2.4"
+RUN bash -l -c " \
+    nvm install 0.8 && \
+    nvm install 0.10 && \
+    nvm install 0.12 && \
+    nvm install iojs-v3.3.1 && \
+    nvm install 4.2 && \
```
Comment (fjakobs) [inline workspace/Dockerfile]: default should be 4.3

File: workspace/Dockerfile [hunk 2]
```diff
@@ -29,65 +34,46 @@ RUN apt-get install -y bash console-setup sudo wget curl git python man-db && \
...
     echo '[ -s "/home/ubuntu/.nvm/nvm.sh" ] && . "/home/ubuntu/.nvm/nvm.sh" # This loads nvm' >> /home/ubuntu/.profile
-RUN sudo -u ubuntu -i bash -l -c " \
-        nvm install 0.8.28 && \
-        nvm install 0.10.41 && \
-        nvm install 0.12.9 && \
-        nvm install iojs-v3.3.1 && \
-        nvm install 4.2.4 && \
-        nvm install 5.4.1 && \
-        nvm alias default 4.2.4"
+RUN bash -l -c " \
```
Comment (fjakobs) [inline workspace/Dockerfile]: lets only install 4.3 LTS here and have the other ones in the node.js image

File: workspace/Dockerfile [hunk 3]
```diff
@@ -29,65 +34,46 @@ RUN apt-get install -y bash console-setup sudo wget curl git python man-db && \
...
-RUN apt-get install -y texlive
-# RUN apt-get install -y texlive-full
-
 # editors
-RUN apt-get install -y vim vim-addon-manager vim-vimoutliner vim-doc
-RUN apt-get install -y emacs ctags indent
+RUN apt-get install -y nano vim vim-addon-manager vim-vimoutliner vim-doc ctags indent
 
 # X11
 RUN apt-get install -y xvfb tightvncserver x11vnc novnc xterm fluxbox
```
Comment (fjakobs) [inline workspace/Dockerfile]: maybe strip X11, too

File: ws-android/Dockerfile [hunk 4]
```diff
@@ -9,61 +9,3 @@ ADD ./files/workspace /home/ubuntu/workspace
 
 RUN chmod -R g+w /home/ubuntu/workspace && \
     chown -R ubuntu:ubuntu /home/ubuntu/workspace
-
```
Comment (fjakobs) [inline ws-android/Dockerfile]: why did this get removed?

File: ws-nodejs/Dockerfile [hunk 5]
```diff
@@ -1,9 +1,9 @@
-FROM cloud9/workspace
+FROM cloud9/workspace
 MAINTAINER Cloud9 IDE, inc. <info@c9.io>
-
-RUN rm -rf /home/ubuntu/workspace && \
-    env GIT_SSL_NO_VERIFY=true git clone https://github.com/c9/chat-example.git /home/ubuntu/workspace
-    
-RUN chown -R ubuntu:ubuntu /home/ubuntu/workspace && \
-    sudo -u ubuntu -i bash -l -c "cd /home/ubuntu/workspace && npm install" && \
+
```
Comment (fjakobs) [inline ws-nodejs/Dockerfile]: nvm install the node versions from the base image

File: ws-ruby/Dockerfile [hunk 6]
```diff
@@ -6,7 +6,24 @@ RUN echo "Version 1.0"
...
+    NOKOGIRI_USE_SYSTEM_LIBRARIES=1 gem install rails -v 5.0.0.beta3"
+RUN bash -l -c "rvm install ruby-2.2.4 && \
+    rvm use 2.2.4 --create && \
+    NOKOGIRI_USE_SYSTEM_LIBRARIES=1 gem install rails -v 4.2.5"
+RUN bash -l -c "rvm install ruby-2.1.5 && \
+     rvm use 2.1.5 --create && \
+     NOKOGIRI_USE_SYSTEM_LIBRARIES=1 gem install rails -v 4.2.5"
+RUN bash -l -c "rvm install ruby-1.9.3-p551 && \
+    rvm use 1.9.3 --create && \
+    NOKOGIRI_USE_SYSTEM_LIBRARIES=1 gem install rails -v 3.2.22"
```
Comment (fjakobs) [inline ws-ruby/Dockerfile]: I would put the oldest version first. We are more likely to update the newer versions

**Comment by fjakobs** [comment]
Great work. Left a few comments.

File: workspace/Dockerfile [hunk 7]
```diff
@@ -118,130 +92,52 @@ RUN sudo -u ubuntu -i bash -l -c " \
...
-# Alternative python versions
-RUN yes | add-apt-repository ppa:fkrull/deadsnakes && \
-    apt-get update && \
-    apt-get install -y python3.5-complete
-
-# Julia
-RUN apt-get install -y julia julia-doc
+RUN sudo -u ubuntu -i heroku
+RUN sudo -u ubuntu -i bash -l -c "npm install azure-cli -g && \
+                                  gem install rhc -v 1.38.4"
```
Comment (fjakobs) [inline workspace/Dockerfile]: we also have this in shared

**Comment by fjakobs** [comment]
lgtm

**Pull request merged**